projects/igniteui-angular/src/lib/directives/drag-drop/drag-drop.directive.ts

```diff
@@ -909,7 +909,7 @@ export class IgxDragDirective implements AfterContentInit, OnDestroy {
 
         // Set pointer capture so we detect pointermove even if mouse is out of bounds until ghostElement is created.
         const handleFound = this.dragHandles.find(handle => handle.element.nativeElement === event.currentTarget);
-        const targetElement = handleFound ? handleFound.element.nativeElement : this.element.nativeElement;
+        const targetElement = handleFound ? handleFound.element.nativeElement : event.target || this.element.nativeElement;
```

Comment:
@skrustev this is the line that causes the same error to appear again in Webcomponents - 
![image](https://github.com/IgniteUI/igniteui-angular/assets/1182001/c030f779-5e9b-43f9-9ce9-acb7187afcea)

Reply:
@dkamburov This more or less caused the pinning happening on `pointerdown` instead of the usual `click` event. As per the sample, it can be seen that when the pin icon is clicked to pin the column, the focus moves the the `ProductName` column and the pointer down is also triggered for that column, but for some reason the `event.target` remains the icon of the previous column which has been recreated and the current `event.target` is no in the DOM.

I've added a check though in the IgxDragDirective to make sure the target is present in the DOM, so no error should be thrown now. 

But still we should probably fix that in the sample as well when this PR is merged, to pin/unpin the column on `click` instead of `pointerdown`